src/App.tsx

```diff
@@ -1,61 +1,139 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function findUserById(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todoWithUser: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUserById(todo.userId),
+}));
+
+function findUserByName(userName: string): User | null {
+  return usersFromServer.find(user => user.name === userName) || null;
+}
 
 export const App = () => {
+  const [todos, setTodos] = useState(todoWithUser);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState('');
+  const [isTitleError, setIsTitleError] = useState(false);
+  const [isUserNameError, setIsUserNameError] = useState(false);
+
+  const handleSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (selectedUser && title) {
+      setTodos(todosPrev => {
+        const maxTodoId = Math.max(...todosPrev.map(todo => todo.id));
+        const newUser = findUserByName(selectedUser);
+
+        setTitle('');
+        setSelectedUser('');
+
+        return [
+          ...todosPrev,
+          {
+            id: maxTodoId + 1,
+            userId: newUser ? newUser.id : null,
+            title,
+            completed: false,
+            user: newUser,
+          },
+        ];
+      });
+    }
+
+    if (!selectedUser) {
+      setIsUserNameError(true);
+    }
+
+    if (!title) {
+      setIsTitleError(true);
+    }
```

Comment:
Let's make this validation in the top of this function and add return to remove if in addTodo block. Also check how we made validation on the lesson today

Reply:
Changed validation method.